### PR TITLE
Back to Java 7

### PR DESCRIPTION
**Pull request by AlexDBlack**
Fixes: https://github.com/deeplearning4j/nd4j/issues/2352

Tested locally: this builds, and I checked the impacted dependencies via mvn site, looks like everything that is supposed to be java 7 is back to java 7 (nd4j-native and nd4j-cuda still show java 8 in mvn site, but the dependencies are test scope).

### CODE REVIEW

File: nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/Linear.java [hunk 1]
```diff
@@ -43,7 +44,7 @@ public Linear(int nIn,
         super(null,
                 getParams(nIn,nOut,weightInitScheme,biasWeightInitScheme),
                 new INDArray[]{},
```
Comment (agibsonccc) [inline nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/Linear.java]: this is java *6* we should be fine without this.

File: nd4j-backends/nd4j-api-parent/nd4j-api/pom.xml [hunk 2]
```diff
@@ -39,8 +39,8 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <configuration>
-                    <source>1.8</source>
-                    <target>1.8</target>
+                    <source>1.7</source>
```
Comment (agibsonccc) [inline nd4j-backends/nd4j-api-parent/nd4j-api/pom.xml]: yes this makes sense. I want java 8 in the tests now if we can. As for samediff there might have been some modifications there.

File: nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/Linear.java (hunk 1 above)
Comment (AlexDBlack) [reply nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/Linear.java]: I thought so too - but Intellij was complaining about it, so I changed it. I just checked with maven now - it's unhappy there too, without it: https://gist.github.com/AlexDBlack/1f76e216bc26d3946d4bf2061a045e2f

**Comment by saudet** [comment]
@AlexDBlack @agibsonccc
Should we get this done before the coming release?
Or is it OK to have Java 8 as requirement?

**Comment by agibsonccc** [comment]
Yeah..We shouldn't be all in on java 8 *yet*

**Pull request merged**